artifacts/deploy/bootstrap-token-configuration.yaml

```diff
@@ -203,3 +188,79 @@ subjects:
 - apiGroup: rbac.authorization.k8s.io
   kind: Group
   name: system:nodes
+
+# ---
+# apiVersion: rbac.authorization.k8s.io/v1
+# kind: Role
+# metadata:
+#   name: system:karmada:agent-secret
+#   namespace: "{{cluster_namespace}}" # default to karmada-cluster
+# rules:
+# - apiGroups:
+#   - ""
```

Comment:
@B1F030 Can you explain what these roles are and why they are commented out?

Reply:
To minimize the rbac permissions for karmada-agent, we can limit some sensitive permissions to minimum.
Sensitive permissions including:
- clusters: [create, get]
- clusters/status: [update]
- secrets: [get, list, watch, create, patch]
- works: [create, get, list, watch, update, delete]
- works/status: [patch, delete]

For `clusters` and `clusters/status` are not namespace-scoped, we can limit the specific `resourceNames` as `{{clustername}}`.
And for `secrets` and `works`, `works/status` are namespace-scoped, we can collect them into specific roles, as:
`system:karmada:agent-secret`(in namespace {{cluster_namespace}}, default to karmada-cluster) for `secrets`, 
`system:karmada:agent-work`(in namespace karmada-es-{{clustername}}) for `works`, `works/status`.

Reply:
The un-commented parts represent the minimal RBAC permissions that can be achieved without modifying the code and the existing `karmadactl register` workflow.
The commented parts represent the desired minimal RBAC permissions

Reply:
Can we move the commented part into the documentation to demonstrate the agent's minimal permission set, and then later modify the code to align the agent's RBAC permissions with the desired minimal RBAC?